internal/envtest/envtest.go

```diff
@@ -3,69 +3,87 @@ package envtest
 import (
 	"context"
 	"path/filepath"
-	"runtime"
 	"strings"
 	"testing"
 
 	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
 	controlapi "github.com/k8ssandra/cass-operator/apis/control/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-client/pkg/kubernetes"
 	k8ssandrataskapi "github.com/k8ssandra/k8ssandra-operator/apis/control/v1alpha1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
 
-	"github.com/k8ssandra/k8ssandra-client/pkg/kubernetes"
-	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	"k8s.io/kubectl/pkg/scheme"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
 func Run(m *testing.M, setupFunc func(e *Environment)) (code int) {
-	env := NewEnvironment()
-	env.start()
+	ctx := ctrl.SetupSignalHandler()
+	env := NewEnvironment(ctx)
+	env.Start()
 	setupFunc(env)
 	exitCode := m.Run()
-	env.stop()
+	env.Stop()
 	return exitCode
 }
 
 type Environment struct {
-	intClient     client.Client
+	Client        client.Client
 	env           *envtest.Environment
 	cancelManager context.CancelFunc
 	Context       context.Context
+	Kubeconfig    string
 }
 
-func NewEnvironment() *Environment {
+func NewEnvironment(ctx context.Context) *Environment {
 	env := &Environment{}
 	env.env = &envtest.Environment{
-		CRDDirectoryPaths:     []string{filepath.Join(RootDir(), "testfiles", "crd")},
+		CRDDirectoryPaths: []string{
+			filepath.Join(RootDir(), "testfiles", "crd"),
+		},
 		ErrorIfCRDPathMissing: true,
 	}
-
-	ctx := ctrl.SetupSignalHandler()
 	ctx, cancel := context.WithCancel(ctx)
 	env.Context = ctx
 	env.cancelManager = cancel
 	return env
 }
 
-// https://stackoverflow.com/questions/31873396/is-it-possible-to-get-the-current-root-of-package-structure-as-a-string-in-golan
-func RootDir() string {
-	_, b, _, _ := runtime.Caller(0)
-	return filepath.Join(filepath.Dir(b), "../..")
+func (e *Environment) GetClient(namespace string) client.Client {
```

Comment:
issue: Please do not rename things to GetClient, since it's not useful. See https://go.dev/doc/effective_go#Getters for naming conventions in Go. GetClient belongs to Java POJOs and this isn't one.

Reply:
I have so many issues with this part of the codebase:

1. Your GetClient is actually a GetClientInNamespace (which, is in fact, how it is named in K8s, and includes your much-maligned `Get`).
2. Your GetClient (despite being namespaced) does not return this other thing you defined called a [NamespacedClient](https://github.com/Miles-Garnsey/k8ssandra-client/blob/feature/dataplane-registration/pkg/kubernetes/client.go#L15).

All of this needs a refactor IMO. In the interests of moving forward, I've deconflicted the naming by leaving the Client field as is, but changing the GetClient name to GetNamespacedClient - which is reflective of what is actually happening and matches Kubernetes.

I hope that will be acceptable as a short term fix?

Reply:
There's a reason it doesn't return NamespacedClient but client.Client and that's to ensure it is directly usable in calls that require client.Client as their input, but ensures also that the requests do not cross namespace boundaries. That's by design.

Reply:
OK, well I've done what you've asked here I believe.